examples/cutting_stock_problem.rs

```diff
@@ -0,0 +1,179 @@
+
+    // Vector of cutting_patterns, initially populated with the trivial ones that contain exactly
+    // one item. cutting_patterns[i][j] indicates how often item j is in pattern i.
+    let initial_cutting_patterns: Vec<Vec<i32>> = (0..item_sizes.len()).map(|i|
+        (0..item_sizes.len()).map(|j| if i == j { 1 } else { 0 }).collect::<Vec<i32>>()
+    ).collect();
+
+    let mut main_problem = Model::default().minimize();
+
+    let cutting_pattern_vars: Vec<Variable> = initial_cutting_patterns.iter().enumerate().map(|(i, &ref _pattern)| {
+        let pattern = (0..10).map(|x| if x == i { "1" } else { "0" }).collect::<Vec<_>>().join("-");
+        main_problem.add_var(0.0, f64::INFINITY, 1.0, &format!("pattern_{pattern}"), VarType::Integer)
+    }).collect();
+
+    demand.iter().enumerate().for_each(|(i, &count)| {
+        let cons = main_problem.add_cons(
+            cutting_pattern_vars.iter().collect(),
+            &initial_cutting_patterns.iter().map(|pattern| pattern[i] as f64).collect::<Vec<f64>>(),
+            count as f64,
+            f64::INFINITY,
+            &format!("demand_for_item_{i}"));
+        main_problem.set_cons_modifiable(&cons, true);
+    });
+
+    let pricer = CSPPricer {
+        stock_length,
+        item_sizes,
+    };
+
+    main_problem.include_pricer("CSPPricer", "CSPPricer", 1, false, Box::new(pricer));
+    let solved_model = main_problem.solve();
+
+    println!("\nSolution");
+    let solution = solved_model.best_sol().unwrap();
+    for var in solved_model.vars().iter() {
+        let name = var.name();
+        let value = solution.val(var);
+        if value != 0.0 {
+            println!("  {name}={value}")
+        }
+    }
+}
+
+struct CSPPricer<'a> {
+    stock_length: usize,
+    item_sizes: &'a [f64],
+}
+
+impl Pricer for CSPPricer<'_> {
+    fn generate_columns(&mut self, mut model: Model<Solving>, _pricer: SCIPPricer, farkas: bool) -> PricerResult {
+
+        // If the current LP relaxation is infeasible, it is the task of the pricer to generate additional variables that can potentially
+        // render the LP feasible again. This is beyond the scope of this example.
+        assert!(!farkas, "Farkas pricing not supported.");
+
+        let mut pricing_model = Model::default().hide_output().maximize();
+
+        let vars = (0..self.item_sizes.len()).map(|i| {
+            let cons = model.find_cons(&format!("demand_for_item_{i}")).unwrap();
+            let dual_val = model.dual_sol(cons);
+            pricing_model.add_var(0.0, f64::INFINITY, dual_val, &format!("demand_for_item_{i}"), VarType::Integer)
+        }).collect::<Vec<Variable>>();
+
+        pricing_model.add_cons(vars.iter().collect(), self.item_sizes, f64::NEG_INFINITY, self.stock_length as f64, "knapsack_constraint");
+
+        let solved_model = pricing_model.solve();
+
+        let reduced_cost = solved_model.best_sol().map(|sol| 1.0 - sol.obj_val());
+        if reduced_cost.is_some_and(|rc| rc < 0.0) {
```

Comment:
I would do `rc < -1e6` here just to be safe (cycling could happen with numerical inaccuracies)

Reply:
done